control/src/main/java/org/openbase/bco/psc/control/ControlController.java

```diff
@@ -77,52 +106,202 @@
-                    try {
-                        if (controllableObjectRegistry.get(x.getId()).switchPowerState()) {
-                            LOGGER.info("Switched power state of unit " + controllableObjectRegistry.get(x.getId()).getConfig().getLabel() + " with id " + x.getId());
-                        } else {
-                            LOGGER.trace("Did not switch power state of unit " + controllableObjectRegistry.get(x.getId()).getConfig().getLabel() + " with id " + x.getId());
+    public void handleEvent(final Event event) throws InterruptedException {
+        try {
+            LOGGER.trace(event.toString());
+            if (inMultimodalMode) {
+                if (event.getData() instanceof UnitProbabilityCollection) {  // collect intents and handle them
+                    UnitProbabilityCollection unitProbabilityCollection = (UnitProbabilityCollection) event.getData();
+
+                    rsbConnection.publishData(unitProbabilityCollection);
+
+                    List<UnitProbability> selectedUnits = unitProbabilityCollection.getElementList().stream()
+                            .filter(x -> x.getProbability() >= threshold)
+                            .collect(Collectors.toList());
+
+                    if (selectedUnits.size() > 0) {
+                        unitProbabilityCollection = UnitProbabilityCollection.newBuilder().addAllElement(selectedUnits).build();
+                        selectedUnitIntents.put(event.getMetaData().getReceiveTime(), unitProbabilityCollection);
+                        handleIntents();
+                    }
+                } else if (event.getData() instanceof ActionParameter) {
+                    receivedStatesIntents.put(event.getMetaData().getReceiveTime(), (ActionParameter) event.getData());
+                    handleIntents();
+                }
+            } else {
+                if (event.getData() instanceof UnitProbabilityCollection) {
+                    UnitProbabilityCollection collection = (UnitProbabilityCollection) event.getData();
+                    collection.getElementList().stream().filter(x -> x.getProbability() >= threshold).forEach(x -> {
+                        if (controllableObjectRegistry.contains(x.getId())) {
+                            try {
+                                if (controllableObjectRegistry.get(x.getId()).switchPowerState()) {
+                                    LOGGER.info("Switched power state of unit " + controllableObjectRegistry.get(x.getId()).getConfig().getLabel() + " with id " + x.getId());
+                                } else {
+                                    LOGGER.trace("Did not switch power state of unit " + controllableObjectRegistry.get(x.getId()).getConfig().getLabel() + " with id " + x.getId());
+                                }
+                            } catch (InterruptedException ex) {
+                                // skip run because of interruption
```

Comment:
Please forward interruption ``throw ex;``or recover it at least ``thread.currentthread().interrupt();``